### ISSUE TITLE
Why does number of epochs exceed the ``max_epochs`` variable

### ISSUE THREAD

**Issue by vinhqdang** [jcjohnson/torch-rnn#81]
Hello

For instance, in training I set ``max_epochs=5``, then I will see:

Epoch 5.38/5, i = 20000/21000, loss = ...

How can the epoch be a float number, and also it can exceed the value of ``max_epochs``?

Thanks

**Comment by ChrisCummins** [comment]
The epoch counter starts at 1.0, whereas the `max_epochs` value starts from 0.0. So for with `-max_expochs=5`, it will actually run until Epoch 6/5.
